File: tests/fixtures/_highlight-corpus/sample.erl
%% Erlang sample
-module(sample).
-export([main/1, count_bytes/1]).

-record(counter, {value = 0 :: integer()}).

increment(#counter{value = V}, N) ->
    #counter{value = V + N}.

count_bytes(Path) ->
    {ok, Bin} = file:read_file(Path),
    Lines = binary:split(Bin, <<"\n">>, [global]),
    lists:foldl(fun count_line/2, #counter{}, Lines).

count_line(<<>>, Acc) ->
    Acc;
count_line(Line, Acc) ->
    increment(Acc, byte_size(Line)).

main([]) ->
    main(["input.txt"]);
main([Path | _]) ->
    Counter = count_bytes(Path),
    io:format("total bytes: ~p~n", [Counter#counter.value]).

File: tests/fixtures/_highlight-corpus/sample.ts
// TypeScript sample
import { readFile } from "node:fs/promises";

interface Incrementable {
  increment(n?: number): number;
  readonly value: number;
}

class Counter implements Incrementable {
  private _value = 0;
  increment(n: number = 1): number {
    this._value += n;
    return this._value;
  }
  get value(): number {
    return this._value;
  }
}

async function main(): Promise<void> {
  const data: string = await readFile("input.txt", "utf8");
  const lines: string[] = data.split("\n").filter((l) => l.length > 0);
  const counter: Incrementable = new Counter();
  for (const line of lines) {
    counter.increment(line.length);
  }
  console.log(`total bytes: ${counter.value}`);
}

main();
